portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,42 +34,45 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<String>();
-    comments.add("Wow, Obi nice website!");
-    comments.add("I love the color scheme");
-    comments.add("We love you Obi!");
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(comments);
-    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
```

Comment:
I think you should call this `commentEntity` because it is more specific than `entity`.